app/models/json_builders/identity_json_builder.rb

```diff
@@ -0,0 +1,104 @@
+class JsonBuilders::IdentityJsonBuilder
+=begin
+(Albert Wang). Meant for the browsers. Parity with Ledger's v3/identity.
+=end
+
+  attr_reader :channel_detail, :publisher_name, :errors
+
+  # Copy of bat-ledger/node_modules/bat-publisher/index.js `providerRE`
+  URL_REGULAR_EXPRESSION = /^([A-Za-z0-9][A-Za-z0-9-]{0,62})#([A-Za-z0-9][A-Za-z0-9-]{0,62}):(([A-Za-z0-9-._~]|%[0-9A-F]{2})+)$/
+
+
+  def initialize(publisher_name:)
+    @errors = []
+    @publisher_name = publisher_name
+    @parsed_publisher_name = URL_REGULAR_EXPRESSION.match(@publisher_name)
+    find_channel_detail
+  end
+
+  def find_channel_detail
+    @channel_detail = if @parsed_publisher_name.present? && @parsed_publisher_name[1] == Channel::YOUTUBE
+      YoutubeChannelDetails.find_by(youtube_channel_id: @parsed_publisher_name[3])
+    elsif @parsed_publisher_name.present? && @parsed_publisher_name[1] == Channel::TWITCH
+      TwitchChannelDetails.find_by(twitch_channel_id: @parsed_publisher_name[3])
+    else
```

Comment:
Consider having a case where @parsed_publisher_name is present does not match youtube or twitch. As it is that would flow to finding the site (and failing) and then the `build_site_identity_json` logic. If that's acceptable a comment about why would help.